src/pybind/cephfs/cephfs.pyx

```diff
@@ -2972,15 +3025,12 @@ class NonRecursiveRmtree:
                     raise OpCanceled('rmtree')
 
                 if de.is_dir():
-                    try:
-                        self.curr_dir.try_rmdir(de.d_name, self.suppress_errors)
-                    except ObjectNotEmpty:
-                        if self.add_dir_to_stack(de.d_name):
-                            # since adding new dir to stack was successful, stop
-                            # traversing the current dir and start traversing
-                            # the new dir that has been freshly added to the
-                            # stack.
-                            break
+                    if self.add_dir_to_stack(de.d_name):
+                        # since adding new dir to stack was successful, stop
+                        # traversing the current dir and start traversing
+                        # the new dir that has been freshly added to the
+                        # stack.
+                        break
```

Comment:
You have a bug here. If you `break` after addign the dir to the stack and **then** check (if statement below) if `self.curr_dir` has any failed ops, you will pop the subdir that you just added.